### ISSUE TITLE
modify_singlept_site_neon.py seems to have trouble finding latest file and creates hidden file for output

### ISSUE THREAD

**Issue by ekluzek** [ESCOMP/CTSM#1437]
### Brief summary of bug

The modify_singlept_site_neon.py seems to have trouble finding the latest file in a directory and creates a hidden file for output.

### General bug information

**CTSM version you are using:** ctsm5.1.dev049

**Does this bug cause significantly incorrect results in the model's science?** No

**Configurations affected:** modify_singlept_site_neon.py 

### Important details of your setup / configuration so we can reproduce the bug

`env CLM_INPUT_TESTS=`pwd`/list ./test_driver.sh -i`

Where list has the smia2 testname in it

The filename created is in the output directory desired, but with the name...

.surfdata_hist_78pfts_CMIP6_simyr2000_YELL_c210726.nc

so it's a hidden file

### Important output or errors that show the problem

> The following files found :
> - /glade/p/cesmdata/cseg/inputdata/lnd/clm2/surfdata_map/NEON/surfdata_hist_78pfts_CMIP6_simyr2000_YELL_c210720.nc
> - /glade/p/cesmdata/cseg/inputdata/lnd/clm2/surfdata_map/NEON/surfdata_hist_78pfts_CMIP6_simyr2000_YELL_c210715.nc
> The latest file is chosen : /glade/p/cesmdata/cseg/inputdata/lnd/clm2/surfdata_map/NEON/surfdata_hist_78pfts_CMIP6_simyr2000_YELL_c210715.nc

**Comment by wwieder** [comment]
@negin513  and @ekluzek can you coordinate to see if this issue has been resolved with updates to scripts in #1539 ?

**Comment by ekluzek** [comment]
OK, I think the hidden file part of this is resolved. It does look correct to me in ctsm5.1.dev069

I think finding the latest file can be an issue, because the mechanism to choose the latest file is based on doing a python glob of the list of files. I think most of the time it will work correctly, but there are exceptions to it, as I think it's actually sorting them by when the files were touched rather than creation. We might just need to add an argument to the call of glob to make sure it's sorting it the way we want it to.

@negin513 could we go over this sometime next week?

**Comment by ekluzek** [comment]
I looked up documentation for glob and it says whether or not it's sorted depends on the filesystem. 

https://docs.python.org/3.9/library/glob.html

So we just need to add a sort after glob is done to make sure the list is sorted.

**Comment by negin513** [comment]
Hi @ekluzek . Thanks for looking into this. 
To summarize the issue for my future self:

In `tools/site_and_regional/modify_singlept_site_neon.py` , we have a capability to find the latest surface dataset for that point. For example if we extracted surface dataset for a point multiple times (using `tools/site_and_regional/run_neon`), it finds the surface dataset the last created surface dataset. Here I used `glob.glob` functionality to list the created files. 
@ekluzek points out that this might be wrong if the file is touched after it is created. 
I agree with him.

I've found the solution is to change this line :
```
https://github.com/ESCOMP/CTSM/blob/57175712e10e443f37818c81b8e14c956f28de34/tools/site_and_regional/modify_singlept_site_neon.py#L256
``` 
to 
```
surf_file = sorted(glob.glob(surf_dir + "/" + sf_name))                                                              
```
This should now sort the files by name and in the filenames we have the time tags (creation times). Therefore, it is also sorted by time.

**Comment by negin513** [comment]
We can go another step further and use the arguments `key=os.path.getmtime` or `key=os.path.getctime` to sort by the time it is latest modified. 
```
surf_file = sorted(glob.glob(surf_dir + "/" + sf_name),  key=os.path.getmtime)  
```

But here again the issue will come up that if the file is touched or modified after creation, the latest modified file will be be chosen. 

Therefore, I think we should sort by name which has the creation time tag and it is technically sorted by creation time. @ekluzek do you also agree?

**Thanks so much @ekluzek for pointing this out.🙏 This might have caused issues in future. ** 

References:
- https://stackoverflow.com/questions/6773584/how-is-pythons-glob-glob-ordered
- The difference between `key=os.path.getmtime` or `key=os.path.getctime` is explained [here](https://stackoverflow.com/questions/17958987/difference-between-python-getmtime-and-getctime-in-unix-system) but I don't think we need either one of them.

**Comment by ekluzek** [comment]
Yes, I agree sorting by name would be the most robust in this case where we have the date in the filename. If that weren't the case, using the creation time might be the best way to sort.

**Comment by negin513** [comment]
Thanks @ekluzek. 🙏

**Comment by negin513** [comment]
This issue is being addressed in PR #1461.

**Comment by ekluzek** [comment]
I like the solution above, and that's implemented in #1461.
